### ISSUE TITLE
VMMEM High CPU after KB5017328

### ISSUE THREAD

**Issue by rommeswi** [microsoft/WSL#8930]
### Version

10.0.22000.978

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.57.1

### Distro Version

22.04

### Repro Steps

Run WSLg app (for example emacs)
Put PC into Hibernate
Return from Sleep

### Expected Behavior

PC recovers from sleep without issues.

### Actual Behavior

CPU quickly goes to 100% load. vmmem uses all CPU remaining from other processes. wsl --shutdown takes several minutes to complete.

This issue has also been reported much earlier in #6982 and #8696 but unlike for the original issues
- the problem for all new posts is caused by KB5017328
- there is no developer response 

So this is a new issue specifically for vmmem issues related to KB5017328. I hope it gets more attention than the large amount of posts in the other two issues.

**Comment by ccarlsonpb** [comment]
Also experiencing this issue. It seems to have started happening after KB5017859 for me. I did install KB5017328 a week ago but I don't think I experienced issues until yesterday after installing KB5017859. WSL completely locks up (vmmem using 90+% CPU) and I can't kill it or run any commands. I have yet to find a workaround. Killing every process that uses WSL (vscode, docker for windows, etc.) doesn't help, "wsl.exe --shutdown" hangs indefinitely, only resolution is to restart my whole computer.

**Comment by SiddharthPant** [comment]
My hunch is this is not because of these updates but something else is triggering the issue. My update history looks like this and started facing issues yesterday 30/09/2022. At that time my last update happened 8 days ago. Since then 2 updates have installed but no resolution.
<img width="662" alt="image" src="https://user-images.githubusercontent.com/10671794/193356332-dcff7cd9-078c-43b3-9bbb-6acf109ff440.png">

Experiencing the same thing. After hibernate Vmmem takes up all CPU, wsl --shutdown stops working and the only workaround is to restart the whole system. I also notice my PC doesn't respect my "Put the computer to sleep" setting in power options, it simply keeps running, which I have noticed in the last 10 days.
What's the correct way to restart Vmmem? It simply says access denied to me when I try ending the task.

**Comment by one1note** [comment]
Exactly the same happening to me after switching users in the latest update KB5017383. 
![afbeelding](https://user-images.githubusercontent.com/64214354/193401251-1a8bdd97-8c38-4b2b-8c41-40a5f1901c1c.png)

**Comment by SiddharthPant** [comment]
Found a way to make wsl work without restarting whole system. Do either one of these:
```
wsl --list --verbose
wsl --terminate Ubuntu-20.04
```
or
```
tasklist /svc /fi "imagename eq svchost.exe" | findstr LxssManager
taskkill.exe /F /PID <pid you get in prev command>
```
This will at least kill the WSL process and allow you to work again.

**Comment by ccarlsonpb** [comment]
My computer has since received the "Windows 11, version 22H2" update, and my WSL2 version was also updated to 5.15.57.1, and the issue seems to have disappeared (so far).

**Comment by dazmc** [comment]
I hope @ccarlsonpb is right as it is killing my machine everytime i try to access WSL distro. I have just deleted and updated WSL again and i can see i am on 5.15.57.1. I have to wait until my machine goes to sleep to see if it breaks things

**Comment by trallnag** [comment]
For me the CPU usages goes up whenever I start VS Code. The moment I close VS Code it's back to idle.

**Comment by dazmc** [comment]
for me as soon as i enter a WSL distro in Windows Terminal Preview it KILLS my windows 11 32GB machine so it is unresponsive - taskbar is dead i can't open anything. Things already open i can just about use until it the machine eventually grinds to a halt and i have to reboot and uninstall wsl2 store package. then i'm back.

**Comment by rommeswi** [comment]
The bug has gotten worse for me. Now I do not even need to have an instance of WSL running for the problem to appear. Perhaps it is also worth mentioning that the process is called vmmemWSL.exe

**Comment by SebTardif** [comment]
This requires from me a daily restart of my laptop. That move me from "Windows is smooth with WSL and vscode integration" to "Oh well, really have to move back to Mac with brew"

**Comment by dazmc** [comment]
still a mess - sometimes i dont' have to even go into WSL it trashes my system. I have tried to kill vmmemWSL.exe as well as some other process - it doesn't work. Sometime i tried to reboot/shutdown, however when i come back it hasn't done either (even though it pretends it has). After a dozen reboots my machine is fine UNTIL i try to open a WSL instance then it kills the machine. I really don't understand what MS have done.

Default Distribution: Ubuntu-2110
Default Version: 2
WSL version: 0.66.2.0
Kernel version: 5.15.57.1
WSLg version: 1.0.42
MSRDC version: 1.2.3401
Direct3D version: 1.606.4
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22621.674

**Comment by alonbl** [comment]
Related-To: #6982

**Comment by Garand-Gary** [comment]
To save restarting your machine, from an elevated prompt try:

`taskkill /F /IM wslservice.exe`

**Comment by rommeswi** [comment]
> My computer has since received the "Windows 11, version 22H2" update, and my WSL2 version was also updated to 5.15.57.1, and the issue seems to have disappeared (so far).

I also just received 22H2 and this indeed seems to fix the issue. I will close this issue and reopen upon reports that this issue persists in 22H2.

**Comment by unixdaddy** [comment]
Not fix for me, still lockups regularly

**Comment by dazmc** [comment]
I have done the following (Windows 11)

- removed all my WSL instances 
- **removed** "windows subsystem for  linux" in Windows features
- left "Virtual Machine Platform" **enabled** in Windows features
- rebooted

**Machine after a while still locked up.**

So then i did the following (WIndows 11)

- removed "Virtual Machine Platform" in Windows features
- rebooted

**Machine functions Brilliantly.**

then i 

- enabled "Virtual Machine Platform" in Windows features
- installed a Linux distro
- rebooted

**Hello Mr Lockup**

Something has happened with Virtual Machine Platform and Some Recent windows updates that makes things unusable on Windows 11.
 
I've removed all WSL/Virtual Machine Platform/LInux distro on my windows 11 machine as i prefer not carrying around a brick for no purpose

I have a separate Windows 10 machine which i'm using and functions fine with WSL/Virtual Machine Platform.

Hopefully Microsoft will fix this

**Comment by rzippo** [comment]
Still a daily occurence for me, on Windows 10 22H2.

**Comment by benreaves** [comment]
This worked for me https://github.com/microsoft/WSL/issues/6982#issuecomment-1474994319

**Comment by 7MinutesDead-Git** [comment]
Wanted to report I'm seeing this same issue randomly, seemingly after sleep and resuming.. possibly after hibernating from low battery, but not sure of exact trigger yet. 

On next startup, vmmem process will run hot at 97-99% CPU usage, and wsl becomes unresponsive in all instances and new wsl processes cannot be started. 

This is running wsl within Windows Terminal, for what it's worth.

Forcefully shutting the wsl process down via running `taskkill /F /IM wslservice.exe` in an elevated terminal tab/process and then running `wsl` again is a working bandaid fix for me, like what Garand-Gary mentioned https://github.com/microsoft/WSL/issues/8930#issuecomment-1278624739

**Comment by naml3i** [comment]
> Wanted to report I'm seeing this same issue randomly, seemingly after sleep and resuming.. possibly after hibernating from low battery, but not sure of exact trigger yet.
> 
> On next startup, vmmem process will run hot at 97-99% CPU usage, and wsl becomes unresponsive in all instances and new wsl processes cannot be started.
> 
> This is running wsl within Windows Terminal on Windows 10, for what it's worth.
> 
> Forcefully shutting the wsl process down via running `taskkill /F /IM wslservice.exe` in an elevated terminal tab/process and then running `wsl` again is a working bandaid fix for me, like what Garand-Gary mentioned [#8930 (comment)](https://github.com/microsoft/WSL/issues/8930#issuecomment-1278624739)

Thanks a lot for this! I was quite frustrated in the last few days because for me even Task Manager betrayed me (access denied when select "End task", wsl --shutdown hangs definitively, the only solution was to reboot the machine altogether! 
Based on the date, after a year seems that there's no improvement on this issue. It was closed as well as people opening a lot of similar issue.

**Comment by ludekvodicka** [comment]
Any update for this? It seems this is 3-years long issue and still without any proper solution (except taskill)

**Comment by naml3i** [comment]
In my case both sleeping and hibernating cause this problem. Waking up from hibernating is the worst since 100% CPU usage and  WSL completely hangs.  Waking up from sleeping is slightly better as CPU "only" shoots up to ~50% to keep me warm in the coming winter days.
Restarting K8s cluster or Docker desktop doesn't help: you'll have to wsl --shutdown to completely restart the service. After restarting the vmmem CPU usage fluctuates around 1-3% in idle state and that should be the normal number.

Note well that this is not related to Docker & Docker desktop and its integrated K8s. I tried other local cluster as well (kind, minikube together with Podman, same problem, it's all with WSL machine)

**Comment by rickterry2** [comment]
taskkill /F /IM wslservice.exe

**Comment by rickterry2** [comment]
taskkill /F /IM wslservice.exe 
This cmd is not work for me. On my laptop goto TaskManager -> Service -> WslService and restart 'WslService' manually,  then CPU usage will drop and wsl back to normal.
